backends/dpdk/dpdkArch.cpp

```diff
@@ -2253,4 +2253,63 @@ void CollectAddOnMissTable::postorder(const IR::MethodCallStatement *mcs) {
     return;
 }
 
+bool ElimHeaderCopy::isHeader(const IR::Expression* e) {
+    auto type = typeMap->getType(e);
+    if (type)
+        return type->is<IR::Type_Header>() && !e->is<IR::MethodCallExpression>();
+    return false;
+}
+
+const IR::Node* ElimHeaderCopy::preorder(IR::AssignmentStatement* as) {
+    if (isHeader(as->left) && isHeader(as->right)) {
+        if (auto path = as->left->to<IR::PathExpression>()) {
+            replacementMap.insert(std::make_pair(path->path->name.name,
```

Comment:
what is happening here?

Reply:
It's for populating replacement map by keeping temporary header name as key and source of assignment statement as value.
i.e.
for below assignment statement
 eth_0 = hdr.ethernet
replacement map will be like
replacementMap["eth_0"] = hdr.ethernet;

later on all uses of eth_0 will be replace with there value in replacementMap.

Reply:
I would add this information as a comment in the header file where the replacementMap is defined.